source/error.c

```diff
@@ -30,6 +30,5 @@ static const struct aws_error_info_list error_info_list = { .error_list = error_
                                                             .count      = sizeof(error_info) / sizeof(error_info[0]) };
 
 void aws_cryptosdk_load_error_strings() {
-    aws_load_error_strings();
```

Comment:
On removing this API call to `aws_load_error_strings` the `aws_cryptosdk_load_error_strings()` function  will loose its meaning and ``aws_cryptosdk_load_error_strings` is used pretty extensively for error reporting in the code-base.

Reply:
added
struct aws_allocator *allocator = aws_default_allocator();
    aws_common_library_init(allocator);
because of this comment in the tests of aws_load_error_strings_test
static int s_aws_load_error_strings_test(struct aws_allocator *allocator, void *ctx) {
    (void)ctx;

    /* Load aws-c-common's actual error info.
     * This will fail if the error info list is out of sync with the error enums. */
    aws_common_library_init(allocator);
    return AWS_OP_SUCCESS;
}

Reply:
I see, seems like `aws_common_library_init(allocator);` sets up the 
`aws_register_error_info(&s_list);` which is duplicated in [line 37](https://github.com/aws/aws-encryption-sdk-c/pull/451/files#diff-055331d7f53065375ed31cb9bf00f9beR37) of this function. We probably want to get rid of the duplication in line 37. 

Also, I reckon we should also call [`aws_common_library_clean_up(void)`](https://github.com/awslabs/aws-c-common/blob/master/source/common.c#L263) for clean-up of the init.

Reply:
I think the library initialization should be handled by the user ? like in aws cpp api libraries where the user is responsible for the life cycle ? we could just add some helper functions. I just know using other common libraries like libCURL etc the user is responsible for life cycle since there is not real way of knowing when to initialize or destroy

Reply:
Ive looked at the aws_register_error_info function and it requires absolutely no initialization so there is not need to intialize the library here - all internal structures are static and the error info is stored in a hash of the error_code member. It can be removed later but you dont have to if the codes collide with another error registration the last of them will overwrite the other. so I will revert the change

Reply:
Right, that sounds good to me, thank-you for contribution and re-iterating on this PR.